prepare/metrics/grouped_instance_metrics.py

```diff
@@ -0,0 +1,553 @@
+from copy import deepcopy
+
+import numpy as np
+
+from src.unitxt import add_to_catalog
+from src.unitxt.metrics import (
+    FixedGroupMeanAccuracy,
+    FixedGroupMeanBaselineAccuracy,
+    FixedGroupMeanBaselineStringContainment,
+    FixedGroupMeanParaphraseAccuracy,
+    FixedGroupMeanParaphraseStringContainment,
+    FixedGroupMeanStringContainment,
+    FixedGroupNormCohensHParaphraseAccuracy,
+    FixedGroupNormCohensHParaphraseStringContainment,
+    FixedGroupNormHedgesGParaphraseAccuracy,
+    FixedGroupNormHedgesGParaphraseStringContainment,
+    FixedGroupPDRParaphraseAccuracy,
+    FixedGroupPDRParaphraseStringContainment,
+    GroupMeanAccuracy,
+    GroupMeanStringContainment,
+    GroupMeanTokenOverlap,
+)
+from src.unitxt.test_utils.metrics import test_metric
+
+predictions = [
+    "A B",
+    "BC D",
+    "C",
+    "123",
+    "BCD",
+    10,
+    "  BD",
+    "AB",
+    "I am a dog",
+    "AB C",
+    "AB 1",
+    "GMA",
+    0.123,
+    "BD",
+    "abc",
+]
+
+references = [
+    ["B", "AB", "A"],
+    ["A", "BC D", "BC DF"],
+    ["c", " C"],
+    [13, 23, 234],
+    ["  ", " BD", " BDA"],
+    [1, 10, 100],
+    ["A", "B", "BD"],
+    ["ABC", "ab", "BC"],
+    ["I am a person", "I AM A DOG", "ABC"],
+    ["AB CD", "AB", "ab"],
+    ["AB 1", "AB1"],
+    [" GMA 123", "GMA"],
+    ["123", 0.12],
+    ["BDE", "BCE", "bdefs"],
+    [" abcdefg", "AB", "abcd"],
+]
+
+# possibly multi-column group identifier; 'ignore' is unused
+# use deepcopy so that dicts in list are independent and can be updated separately
+additional_inputs = (
+    [deepcopy({"group": "grp1", "id": 0, "ignore": 1}) for _ in range(5)]
+    + [deepcopy({"group": "grp1", "id": 1, "ignore": 1}) for _ in range(5)]
+    + [deepcopy({"group": "grp2", "id": 0, "ignore": 1}) for _ in range(4)]
+    + [deepcopy({"group": "grp2", "id": 1, "ignore": 0}) for _ in range(1)]
+)
+# for group_mean aggregations with a subgroup_comparison, add a variant_type label
+# these groupings correspond in length to the group identifiers above
+variant_type = np.concatenate(
+    [
+        np.repeat(a=["original", "paraphrase"], repeats=reps)
+        for reps in [[1, 4], [1, 4], [1, 3], [1, 0]]
+    ]
+).tolist()
+
+# construct grouping_field by combining two other fields (and ignoring one); mimics what you would do in cards
+group_by_fields = ["group", "id"]
+
+for ai, vt in zip(additional_inputs, variant_type):
+    ai.update(
+        {
+            "group_id": "_".join([str(ai[ff]) for ff in group_by_fields]),
+            "variant_type": vt,
+        }
+    )
+
+
+instance_targets_string_containment = [
```

Comment:
why is this a dict and not just a list?

Reply:
It is like this for all the other metrics tested (see e.g. accuracy.py), since the receives the whole instance score dict and not just the list of scores.  They check that the key names (i.e., the metric name) match as well.

Reply:
got it, thanks!